### ISSUE TITLE
T_MAX value (CosineAnnealingLR)

### ISSUE THREAD

**Issue by josemerinom** [konstmish/prodigy#10]
Hello, thanks for the optimizer
I have a question
I am training LoRa with kohya-ss/sd-scripts
I have a dataset of 15 images
I repeat x1 the dataset in each epoch (There are people who do more repetitions per epoch, I try 1 repetition)
and I train 70 epochs = 1050 steps 
and I use wandb to see graphs
I know that the value I indicated in t_max is the step where it goes down or up (sorry, I don't know the technical terms)

my question? What value should I give to t_max?
the total steps? (in my test =1050)
the total number of steps in each epoch? (in my test =15, in my case (repetition x1), every 15 steps another epoch begins)

thanks

--lr_scheduler_type "CosineAnnealingLR" \
--lr_scheduler_args "T_max=???" \
--max_train_epochs=70 \
--optimizer_type="Prodigy" \
--optimizer_args "betas=(0.9, 0.999)" "d_coef=1.0" "d0=1e-6" "decouple=True" "eps=1e-8" "use_bias_correction=True" "safeguard_warmup=True" "weight_decay=0 "\
--save_every_n_epochs=10 \
--log_with="wandb" \

**Comment by madman404** [comment]
If you are using SD scripts, it is sufficient to pass the argument `--lr_scheduler="cosine"` for setting the scheduler. It will set the appropriate T_max itself.
As an aside, I would highly recommend you do not train without weight decay. If you are keeping it low to be safe, 0.01 would be a reasonable start.
